SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -371,19 +380,17 @@ void USpatialNetDriver::QueryGSMToLoadMap()
 	GlobalStateManager->QueryGSM(true /*bRetryUntilAcceptingPlayers*/);
 }
 
-void USpatialNetDriver::HandleOngoingServerTravel()
+bool USpatialNetDriver::IsGlobalStateManagerServerTravelReady() const
 {
-	check(!bConnectAsClient);
-
-	// Here if we are a server and this is server travel (there is a snapshot to load) we want to load the snapshot.
-	if (!ServerConnection && !SnapshotToLoad.IsEmpty() && Cast<USpatialGameInstance>(GetWorld()->GetGameInstance())->bResponsibleForSnapshotLoading)
-	{
-		UE_LOG(LogSpatialOSNetDriver, Log, TEXT("Worker authoriative over the GSM is loading snapshot: %s"), *SnapshotToLoad);
-		SnapshotManager->LoadSnapshot(SnapshotToLoad);
+	return GlobalStateManager->GetAcceptingPlayers() &&
+		   (SessionId == 0 || SessionId == GlobalStateManager->GetSessionId());
```

Comment:
Why the check for 0 here? I understand it's the default, but couldn't we just always confirm that our session id matches that of the gsm?

Reply:
Discussed and sorted.